### PR TITLE
Added Teachers#show

### PR DESCRIPTION
**Pull request by justinphyo**
Admin can now view a teacher's information by clicking the teacher's name on the dashboard or all teachers table.

Something I'm wondering is if I should add a link to the show page and a link to the edit page within the table so that the admin doesn't have to go to show page first to go to edit page. I think the table should be cleaned up a bit if we were to add more stuffs on there.

One more thing -> I might have to change the tests to make sure it matches with the newly updated tests from the Microsoft OAuth PR.

Here's a preview
Screen Shot 2021-04-19 at 10 58 55 PM

### CODE REVIEW

**Review by christopher-hou — APPROVED** [review]
Looks really good! I actually like the way this flows currently lol. If you need to revise based off the login stuff then maybe we can merge that one first, then this one.

**Comment by longhtong** [comment]
great work!!!!

**Review by longhtong — APPROVED** [review]

File: app/controllers/teachers_controller.rb [hunk 1]
```diff
@@ -51,6 +51,12 @@ def edit
     @readonly = !is_admin?
   end
 
+  def show
```
Comment (tommywei110) [inline app/controllers/teachers_controller.rb:54]: I could be wrong, but I think you need to add :show to the require_admin precondition on top of this file.

File: app/views/teachers/_teacher_info.html.erb [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        <a href=<%= teacher.school.website %>><%= teacher.school.name %></a>
+      </div>
+
+      <div class="col-sm-6">
+        <h5> School Location </h5>
+        <%= teacher.school.location %>
+      </div>
+    </div>
+  </div>
+</div>
```
Comment (tommywei110) [inline app/views/teachers/_teacher_info.html.erb]: Code style is complaining about a lack of newline for a few of the files.

File: app/views/teachers/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<%= provide(:h1, "#{@status} View #{@teacher.full_name}") %>
+
+<%= render 'teacher_info', teacher: @teacher %>
```
Comment (tommywei110) [inline app/views/teachers/show.html.erb]: \n newline

**Review by tommywei110 — CHANGES_REQUESTED** [review]
The page looks awesome. Only thing is I think the show function should require admin in the controller.

File: app/views/teachers/_teacher_info.html.erb (hunk 2 above)
Comment (justinphyo) [reply app/views/teachers/_teacher_info.html.erb]: Thanks for the catch. I fixed it.

File: app/views/teachers/show.html.erb (hunk 3 above)
Comment (justinphyo) [reply app/views/teachers/show.html.erb]: Fixed this one too.

**Review by tommywei110 — APPROVED** [review]
Great Work!

**Comment by christopher-hou** [comment]
Merged this in since people are still looking at Microsoft Auth. I'll update the tests to match in the other PR

**Pull request merged**